File: src/app/src/main/java/com/huawei/training/java/fragments/ExamFragment.java
/*
 * Copyright 2020. Huawei Technologies Co., Ltd. All rights reserved.
 *
 *    Licensed under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License.
 *    You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software
 *    distributed under the License is distributed on an "AS IS" BASIS,
 *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *    See the License for the specific language governing permissions and
 *    limitations under the License.
 */

package com.huawei.training.java.fragments;

import android.annotation.SuppressLint;
import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.fragment.app.Fragment;

import com.huawei.training.java.activities.BaseActivity;
import com.huawei.training.java.activities.ExamActivity;
import com.huawei.training.databinding.FragmentExamBinding;
import com.huawei.training.java.models.ExamModel;
import com.huawei.training.java.models.UserObj;
import com.huawei.training.java.utils.LearningApplication;
import com.huawei.training.java.utils.eventmanager.AppAnalytics;
import com.huawei.training.java.utils.video.Constants;

import org.jetbrains.annotations.NotNull;

import java.util.Objects;

/**
 * @since 2020
 * @author Huawei DTSE India
 */
public class ExamFragment extends Fragment {
    private static final String ARG_PARAM1 = "param1";
    /**
     * The Binding.
     */
    FragmentExamBinding binding;
    /**
     * The Exam model.
     */
    ExamModel examModel = null;
    /**
     * The App analytics.
     */
    AppAnalytics appAnalytics = null;
    /**
     * The User obj.
     */
    UserObj userObj = null;

    /**
     * Instantiates a new Exam fragment.
     */
    public ExamFragment() {
    }

    /**
     * New instance exam fragment.
     *
     * @param examModel the exam model
     * @return the exam fragment
     */
    public static ExamFragment newInstance(ExamModel examModel) {
        ExamFragment fragment = new ExamFragment();
        Bundle args = new Bundle();
        args.putParcelable(ARG_PARAM1, examModel);
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            examModel = getArguments().getParcelable(ARG_PARAM1);
        }
        appAnalytics = ((BaseActivity) Objects.requireNonNull(getActivity())).appAnalytics;
        userObj = ((LearningApplication) getActivity().getApplication()).getUserObj();
    }

    @SuppressLint("SetTextI18n")
    @Override
    public View onCreateView(@NotNull LayoutInflater inflater,
                             ViewGroup container, Bundle savedInstanceState) {
        binding = FragmentExamBinding.inflate(inflater, container, false);
        binding.examTitle.setText(examModel.getName());
        binding.examDesc.setText(examModel.getDesc());
        binding.examDurationMin.setText("" + examModel.getDuration());
        binding.examTotalQC.setText("" + examModel.getNoOfQuestions());

        binding.startexam.setOnClickListener(start -> {
                    if (!((BaseActivity) Objects.requireNonNull(getActivity())).showLoginDialog()) {
                        if (userObj == null)
                            userObj = ((LearningApplication) getActivity()
                                    .getApplication()).getUserObj();
                        if (userObj != null) {
                            navigatetoExam();
                        }
                    }
                }
        );
        return binding.getRoot();
    }

    // navigate to exam activity
    private void navigatetoExam() {
        appAnalytics.startExamClickEvent(examModel.getName());
        Intent intent = new Intent(getActivity(), ExamActivity.class);
        intent.putExtra(Constants.COURSE_NAME, examModel.getName());
        intent.putExtra(Constants.COURSE_ID, "" + examModel.getCId());
        startActivity(intent);
    }
}

File: src/app/src/main/java/com/huawei/training/kotlin/fragments/ExamFragment.kt
/*
 * Copyright 2020. Huawei Technologies Co., Ltd. All rights reserved.
 *
 *    Licensed under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License.
 *    You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software
 *    distributed under the License is distributed on an "AS IS" BASIS,
 *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *    See the License for the specific language governing permissions and
 *    limitations under the License.
 */
package com.huawei.training.kotlin.fragments

import android.annotation.SuppressLint
import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.huawei.training.kotlin.activities.BaseActivity
import com.huawei.training.kotlin.activities.ExamActivity
import com.huawei.training.databinding.FragmentExamBinding
import com.huawei.training.kotlin.models.ExamModel
import com.huawei.training.kotlin.models.UserObj
import com.huawei.training.kotlin.utils.LearningApplication
import com.huawei.training.kotlin.utils.eventmanager.AppAnalytics
import com.huawei.training.kotlin.utils.video.Constants
import java.util.*

/**
 * @since 2020
 * @author Huawei DTSE India
 */
class ExamFragment
/**
 * Instantiates a new Exam fragment.
 */
    : Fragment() {
    /**
     * The Binding.
     */
    var binding: FragmentExamBinding? = null

    /**
     * The Exam model.
     */
    var examModel: ExamModel? = null

    /**
     * The App analytics.
     */
    var appAnalytics: AppAnalytics? = null

    /**
     * The User obj.
     */
    var userObj: UserObj? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if (arguments != null) {
            examModel = arguments!!.getParcelable(ARG_PARAM1)
        }
        appAnalytics = (Objects.requireNonNull(activity) as BaseActivity).appAnalytics
        userObj = (activity!!.application as LearningApplication).userObj
    }

    @SuppressLint("SetTextI18n")
    override fun onCreateView(inflater: LayoutInflater,
                              container: ViewGroup?, savedInstanceState: Bundle?): View? {
        binding = FragmentExamBinding.inflate(inflater, container, false)
        binding?.examTitle?.text = examModel!!.name
        binding?.examDesc?.text = examModel!!.desc
        binding?.examDurationMin?.text = "" + examModel!!.duration
        binding?.examTotalQC?.text = "" + examModel!!.noOfQuestions
        binding?.startexam?.setOnClickListener { start: View? ->
            if (!(Objects.requireNonNull(activity) as BaseActivity).showLoginDialog()) {
                if (userObj == null) userObj = (activity
                        ?.getApplication() as LearningApplication).userObj
                if (userObj != null) {
                    navigatetoExam()
                }
            }
        }
        return binding?.root
    }

    // navigate to exam activity
    private fun navigatetoExam() {
        appAnalytics!!.startExamClickEvent(examModel!!.name)
        val intent = Intent(activity, ExamActivity::class.java)
        intent.putExtra(Constants.COURSE_NAME, examModel!!.name)
        intent.putExtra(Constants.COURSE_ID, "" + examModel!!.cId)
        startActivity(intent)
    }

    companion object {
        private const val ARG_PARAM1 = "param1"

        /**
         * New instance exam fragment.
         *
         * @param examModel the exam model
         * @return the exam fragment
         */
        fun newInstance(examModel: ExamModel?): ExamFragment {
            val fragment = ExamFragment()
            val args = Bundle()
            args.putParcelable(ARG_PARAM1, examModel)
            fragment.arguments = args
            return fragment
        }
    }
}